### DISCUSSION TITLE
Are there any example of how to Activate a Window in unpackaged app?

### DISCUSSION THREAD

**Discussion by shelllet** [microsoft/WindowsAppSDK#1692] [Q&A] [answered]
Are there any example of how to Activate a Window in unpackaged app?

Is it so?

```
MddBootstrapInitialize(...)

auto window = make<MainWindow>();
window Activate();

// where is message loop ?

MddBootstrapShutdown();
```

**Comment by andrewleader** [comment, marked as answer]
Hi @shelllet, we'll need some more info in order to help you.

Are you trying to build a WinUI 3 unpackaged app? If so, [the docs](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/create-your-first-winui3-app?tabs=csharp%2Ccsharp-unpackaged#instructions-for-winui-3-unpackaged-desktop-apps) walk through this and the File->New template includes the code for activating the window.

If you're trying to do something else, we'll need to know exactly what you're trying to do. In 1.0, mixing WinUI 3 and other UI stacks is not officially supported, so you can't create and activate a WinUI 3 window from another type of app.

**Reply by shelllet** [reply]
thks,  **build a WinUI 3 unpackaged app**  this is what i mean. the docs [Tutorial: Build and deploy an unpackaged app that uses the Windows App SDK](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/tutorial-unpackaged-deployment?tabs=cpp) tell me how to build and deploy unpackaged  app,but did not tell us how to mix winui3  and activate window in main function.

in official example:

```
#include <iostream> 
#include <windows.h> 
#include <MddBootstrap.h>

int main() 
{ 

    // Take a dependency on Windows App SDK 1.0 Preview 3. If using 1.0 Experimental,
    // replace with versionTag{ L"experimental1" }. If using version 0.8 Preview, 
    // replace with majorMinorVersion{ 0x00000008 } and  versionTag{ L"preview" }.
    const UINT32 majorMinorVersion{ 0x00010000 }; 
    PCWSTR versionTag{ L"preview3" }; 
    const PACKAGE_VERSION minVersion{};

    const HRESULT hr{ MddBootstrapInitialize(majorMinorVersion, versionTag, minVersion) }; 

    // Check the return code. If there is a failure, display the result.
    if (FAILED(hr)) 
    { 
        wprintf(L"Error 0x%X in MddBootstrapInitialize(0x%08X, %s, %hu.%hu.%hu.%hu)\n", 
            hr, majorMinorVersion, versionTag, minVersion.Major, minVersion.Minor, minVersion.Build, minVersion.Revision); 
        return hr; 
    } 

    std::cout << "Hello World!\n"; 

    // Release the DDLM and clean up.
    MddBootstrapShutdown(); 
}
```

**Reply by andrewleader** [reply]
Ah, that's some older documentation before we had full WinUI 3 unpackaged support. Here's the doc you should be using: https://docs.microsoft.com/en-us/windows/apps/winui/winui3/create-your-first-winui3-app?tabs=csharp%2Ccpp-unpackaged#instructions-for-winui-3-unpackaged-desktop-apps

Let me know if you have any further questions!

**Reply by shelllet** [reply]
This may be different from what I thought。create normal `win32  desktop app`  by visual studio `win32 desktop template` and change it to winui3 window seems not possible.

**Reply by andrewleader** [reply]
You could probably do that, but it would involve a lot of work. Is starting with a new WinUI 3 C++ app using the WinUI 3 C++ template not possible? If so, why?

**Reply by shelllet** [reply]
thks your explaination, i just want to ask if it is possible, or if there are examples, which may be useful in some cases, to be honest, I don’t really do that.  and i don't need to do that now

**Reply by andrewleader** [reply]
We don't have any examples of starting with a win32 desktop app and changing it to use WinUI 3, we recommend starting with a WinUI 3 project template as that sets up everything correctly from the start. It's probably possible, but unless there's a specific business need, it's easier to just start with the WinUI 3 project template.
